components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -2000,4 +2008,215 @@ public static SAMLSSOServiceProviderDO getSPConfig(String tenantDomain, String i
             PrivilegedCarbonContext.endTenantFlow();
         }
     }
+
+    /**
+     * Build SAML logout request.
+     *
+     * @param serviceProviderDO SP for which the logout request is built.
+     * @param subject           Subject identifier.
+     * @param sessionId         Session index.
+     * @return Logout Request.
+     * @throws IdentityException If tenant domain is invalid.
+     */
+    public static LogoutRequest buildLogoutRequest(SAMLSSOServiceProviderDO serviceProviderDO, String subject,
```

Comment:
better if we can follow the same pattern as the authentication response builder.